src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java

```diff
@@ -0,0 +1,222 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+
+/**
+ * This implementation checks that keys of properties files are sorted.
+ * It is using the binary order from String.CASE_INSENSITIVE_ORDER.compare()
+ * The sort direction is ascending.
+ *
+ *   <p>
+ *       Example properties file:
+ *   </p>
+ *   <pre>
+ *       &#64;=64 (it is the 'at' sign)
+ *       A=65
+ *       a=97
+ *       key=107 than nothing
+ *       key.sub=k is 107 and dot is 46
+ *   </pre>
+ *   <ul>
+ *       <li>
+ *           The &#64; sign is on position 64 and comes first.
+ *       </li>
+ *       <li>
+ *           The capital A is on 65 and the lowercase a is on position 97 on the ascii table.
+ *       </li>
+ *       <li>
+ *           Key and subkey are in correct order here, because only keys are relevant.
+ *           Therefore on line 5 you have only "key" an nothing behind.
+ *           On line 6 you have "key." The dot is on position 46 which is higher than nothing.
+ *       </li>
+ *   </ul>
+ */
+@StatelessCheck
+public class OrderedPropertiesCheck extends AbstractFileSetCheck {
+
+    /**
+     * Localization key for check violation.
+     */
+    public static final String MSG_KEY = "properties.notSorted.property";
+    /**
+     * Localization key for IO exception occurred on file open.
+     */
+    private static final String UNABLE_TO_OPEN_FILE = "Unable to open file: ";
+    /**
+     * Pattern matching single space.
+     */
+    private static final Pattern SPACE_PATTERN = Pattern.compile(" ");
+
+    /**
+     * Construct the check with default values.
+     */
+    public OrderedPropertiesCheck() {
+        setFileExtensions("properties");
+    }
+
+    /**
+     * Processes the file and check order.
+     * @param file the file to be processed
+     * @param fileText the contents of the file.
+     * @throws CheckstyleException
+     * I like to keep the order - keySet is not possible
```

Comment:
This doesn't seem to be the reasoning for a checkstyle exception. Please update.

Reply:
Done.

Reply:
@thomassenger
> I like to keep the order - keySet is not possible

This shouldn't be connected to the `@param` either. If there is no next `@` the text goes with the previous when it is printed to the javadoc.
Either put it after the full `@noinspection` or remove it.